src/scripts/views/form/index.js

```diff
@@ -1,79 +1,117 @@
-			type: React.PropTypes.oneOf([
-				'text', 'email', 'password'
-			]).isRequired,
-			name:     React.PropTypes.string.isRequired,
-			label:    React.PropTypes.string.isRequired,
-			title:    React.PropTypes.string,
-			pattern:  React.PropTypes.string,
-			required: React.PropTypes.bool
-		})),
-
-		secondary: React.PropTypes.shape({
-			submit:      React.PropTypes.func.isRequired,
-			action:      React.PropTypes.string.isRequired,
-			description: React.PropTypes.string
-		}),
-
-		help:   React.PropTypes.string,
-		submit: React.PropTypes.func.isRequired,
-		action: React.PropTypes.string.isRequired
-	},
-
-	getDefaultProps() {
-		return { fields: [ ], secondary: null }
+		formProfile: React.PropTypes.string.isRequired,
+		boardID: React.PropTypes.string,
+		accessCode: React.PropTypes.string
 	},
 
 	getInitialState() {
-		return this.props.fields.reduce((state, field) => {
+		return FormData.registerForm.fields.reduce((state, field) => {
 			state[field.name] = '';
 			return state;
 		}, {});
 	},
 
-	submitPrimary(event) {
-		this.props.submit(this.state);
-		return event.preventDefault();
+	checkPasswords(){
+		if(this.props.formProfile === 'registerForm' && this.state.passwordAgain !== '') {
+			return this.state.passwordAgain !== this.state.passwordRegister ?
+				<span className="fa fa-times mismatch">Password mismatch!</span>
+				: <span className="fa fa-check match">Passwords match!</span>;
+		}
 	},
 
-	submitSecondary(event) {
-		this.props.secondary.submit(this.state);
-		return event.preventDefault();
+	renderFields(fields) {
+		return fields.map((field, index) => {
+			let controlattrs = {
+				title:    field.title,
+				pattern:  field.pattern,
+				required: field.required
+			}
+			return (
+				<section key={field.name} className="input">
+					<label htmlFor={field.name}>{field.label}</label>
+					<input autoFocus={index === 0} name={field.name}
+						type={field.type} {...controlattrs}
+						valueLink={this.linkState(field.name)} />
+				</section>
+			);
+		});
+	},
+	//submit will execute in all cases other than when
+	//passwords given in registration do not match
+	submitPrimary(currentForm) {
+		if(this.props.formProfile !== 'registerForm' ||
+			this.state.passwordAgain === this.state.passwordRegister) {
+			return (event) => {
+				if(this.props.formProfile === 'registerForm')
+					this.state.password = this.state.passwordRegister;
+				currentForm.submit(this.state);
+				return event.preventDefault();
+			}
+		}
+		else return (event) => {
+			BroadcastAction.add({
+				type:    'Error',
+				content: 'Passwords entered do not match!'
+			});
+			return event.preventDefault();
+		}
 	},
 
-	render() {
-		let secondaryContent = !this.props.secondary ? null : (
+	submitSecondary(currentForm) {
+		return (event) => {
+			currentForm.secondary.submit(this.state);
+			return event.preventDefault();
+		}
+	},
+
+	submitGuest(currentForm, accessCode, boardID){
```

Comment:
I don't think you need a separate function for submitting the guest form, you can use the form arguments as `...rest` parameters, and then use something like `Function.prototype.apply` to apply the arguments to the `submit` function.